.github/workflows/ci.yaml

```
@@ -47,6 +47,48 @@ jobs:
       - run: npm ci
       - run: npm run typecheck
 
+  validate-blog:
+    if: startsWith(github.head_ref, 'blog/')
+    needs: [build, lint, typecheck]
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .node-version
+          cache: npm
+      - run: npm ci
+      - run: git diff origin/main..HEAD --name-only | xargs npm run validate-blog --
+
+  update-blogmeta:
+    needs: [validate-blog]
+    permissions:
+      pages: write
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: .node-version
+          cache: npm
+      - run: npm ci
+      - name: Update blog meta
+        id: update_blogmeta
+        run: |
+          git diff origin/main..HEAD --name-only | xargs npm run update-blogmeta --
+          echo "{UPDATE_RESULT}_{$?}" >> "$GITHUB_OUTPUT"
+        continue-on-error: true
+      - run: exit ${{ steps.update_blogmeta.outputs.UPDATE_RESULT }}
+        if: steps.update_blogmeta.outputs.UPDATE_RESULT != '1' && steps.update_blogmeta.outputs.UPDATE_RESULT != '0'
+      - name: Git Commit
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
```

BotのユーザーIDを入れると良いらしいです https://github.com/orgs/community/discussions/26560#discussioncomment-3252339

```suggestion
          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
```